### PR TITLE
UKRIW-142 -  UI components part2

### PR DESCRIPTION
**Pull request by KatarzynaPaszko**
ticket:: https://ukri-spyrosoft.atlassian.net/browse/UKRIW-142

This PR contains:

- Select, 
- Input Text, 
- SearchInput
- TextArea

To test it, run storybook and compare [with designs](https://www.figma.com/design/WDCJlvNjehFTkHVb4FSulP/UKRI-(original-bid-design)?node-id=2381-62098&t=uQLLWdWsDOEgIlak-1)

### CODE REVIEW

File: libs/shared/design-system/src/lib/select/select.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  return (
+    <div className={selectStyles.container} ref={ref}>
+      <div className='relative'>
+        <button
+          type='button'
+          className={selectStyles.button}
+          aria-haspopup='listbox'
+          aria-expanded={isOpen}
+          aria-labelledby='listbox-label'
+          onClick={() => setIsOpen(!isOpen)}
```
Comment (mlu-ssoft) [inline libs/shared/design-system/src/lib/select/select.tsx:37]: I would move this to `handleClose` function or something. It is hard to read if the logic is hidden deeply in html tree

File: libs/shared/design-system/src/lib/textarea/textarea.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+import textareaStyles from './textarea.styles'; // Adjust the path as needed
```
Comment (mlu-ssoft) [inline libs/shared/design-system/src/lib/textarea/textarea.tsx:3]: what this comment means?

File: libs/shared/design-system/src/lib/textarea/textarea.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  placeholder: string;
+  rows?: number;
+  maxLength?: number;
+}
+
+export const Textarea = ({ label, placeholder, rows = 4, maxLength }: ITextareaProps) => {
+  const [text, setText] = useState('');
+
+  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setText(e.target.value);
```
Comment (mlu-ssoft) [inline libs/shared/design-system/src/lib/textarea/textarea.tsx:16]: won't it cause performance issues?

File: libs/shared/design-system/src/lib/textarea/textarea.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        id='textarea'
+        placeholder={placeholder}
+        rows={rows}
+        maxLength={maxLength}
+        value={text}
+        onChange={handleChange}
+      />
+      {maxLength && (
+        <p className={textareaStyles.charCount}>
+          {text.length}/{maxLength} characters
```
Comment (mlu-ssoft) [inline libs/shared/design-system/src/lib/textarea/textarea.tsx:35]: use translation keys

**Review by mlu-ssoft — CHANGES_REQUESTED** [review]

File: libs/shared/design-system/src/lib/select/select.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  label: string;
+}
+
+interface ISelectProps {
+  options: IOption[];
+  onChange: (option: IOption | null) => void;
+  placeholder?: string;
+}
+
+export const Select = ({ options, onChange, placeholder = 'Select an option' }: ISelectProps) => {
```
[the remark this answers is not in the document]
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/select/select.tsx:17]: Your PR with translations is not merged yet, so can't use translation function

File: libs/shared/design-system/src/lib/input/text-input/text-input.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  className?: string;
+  clearButton?: boolean;
+  iconName?: keyof typeof IconsNames;
+  iconWidth?: number;
+  iconHeight?: number;
+}
+
+export const TextInput: React.FC<ITextInputProps> = ({
+  placeholder = 'Start typing...',
```
[the remark this answers is not in the document]
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/input/text-input/text-input.tsx:19]: Will do once multi language support PR will be merged into develop

File: libs/shared/design-system/src/lib/textarea/textarea.tsx (hunk 2 above)
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/textarea/textarea.tsx:3]: removed

File: libs/shared/design-system/src/lib/textarea/textarea.tsx (hunk 4 above)
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/textarea/textarea.tsx:35]: Will do once multi language support PR will be merged into develop

File: libs/shared/design-system/src/lib/textarea/textarea.tsx (hunk 3 above)
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/textarea/textarea.tsx:16]: I will add useCallback

**Review by mlu-ssoft — APPROVED** [review]

**Pull request merged**